src/util/TypeGuards.js

```diff
@@ -0,0 +1,87 @@
+'use strict';
+
+const { ApplicationCommandTypes, ApplicationCommandOptionTypes, MessageButtonStyles } = require('./Constants');
+
+/**
+ * Helpers for checking different variants of common types.
+ */
+class TypeGuards extends null {
+  /**
+   * Resolves a given type to an enum equivalent value, and
+   * checks if it's part of the given enum type.
+   * @private
+   * @param {string|number} type The type to resolve
+   * @param {any} object The enum to resolve to
+   * @param {any[]} fields The enum fields to check
+   * @returns {boolean} Whether the type is part of the enum
+   */
+  static isPartOfEnum(type, object, fields) {
+    const resolvedType = typeof type === 'number' ? type : object[type];
+    fields = fields.map(field => object[field]);
+    return fields.includes(resolvedType);
```

Comment:
```suggestion
    return fields.some(field => object[field] === resolvedType);
```
*(and remove the above line)*